File: src/Projects/Android/PaymentSystem/PaymentServer/src/main/java/org/csystem/app/generator/text/server/configuration/socket/ServerSocketConfig.java
package org.csystem.app.generator.text.server.configuration.socket;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.io.IOException;
import java.net.ServerSocket;

@Configuration
public class ServerSocketConfig {
    @Bean
    public ServerSocket createServerSocket(@Value("${server.tcp.port}")int port, @Value("${server.tcp.backlog}") int backlog) throws IOException
    {
        return new ServerSocket(port, backlog);
    }
}

File: src/Projects/Android/ChatSystem/Server/src/main/java/org/csystem/app/chat/server/configuration/socket/ServerSocketConfig.kt
package org.csystem.app.chat.server.configuration.socket

import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import java.net.ServerSocket

@Configuration
open class ServerSocketConfig {
    @Bean("app.chat.server.config.register.port.ServerSocket")
    open fun createRegisterServerSocket(
        @Value("\${app.chat.server.config.register.port}") port: Int,
        @Value("\${app.chat.server.config.register.backlog}") backlog: Int
    ): ServerSocket
    {
        return ServerSocket(port, backlog)
    }

    @Bean("app.chat.server.config.login.port.ServerSocket")
    open fun createLoginServerSocket(
        @Value("\${app.chat.server.config.login.port}") port: Int,
        @Value("\${app.chat.server.config.login.backlog}") backlog: Int
    ): ServerSocket
    {
        return ServerSocket(port, backlog)
    }
}
